ui/e2e/explorer/consumer/index.spec.ts

```diff
@@ -0,0 +1,148 @@
+import { createNamespace, deleteNamespace } from "e2e/utils/namespace";
+import { expect, test } from "@playwright/test";
+
+import { createConsumerYaml } from "./utils";
+import { faker } from "@faker-js/faker";
+
+let namespace = "";
+
+test.beforeEach(async () => {
+  namespace = await createNamespace();
+});
+
+test.afterEach(async () => {
+  await deleteNamespace(namespace);
+  namespace = "";
+});
+
+test("it is possible to create a consumer", async ({ page }) => {
+  /* prepare data */
+  const filename = "myconsumer.yaml";
+  const groups = Array.from({ length: 5 }, () => faker.lorem.word());
+  const tags = Array.from({ length: 2 }, () => faker.lorem.word());
+
+  const expectedYaml = createConsumerYaml({
+    username: "my-username",
+    password: "a-v3ry-g00d-pASSw0rd",
+    apiKey: "some-api-key",
+    groups,
+    tags,
+  });
+
+  /* visit page */
+  await page.goto(`/${namespace}/explorer/tree`, { waitUntil: "networkidle" });
+  await expect(
+    page.getByTestId("breadcrumb-namespace"),
+    "it navigates to the test namespace in the explorer"
+  ).toHaveText(namespace);
+
+  /* create consumer */
+  await page.getByRole("button", { name: "New" }).first().click();
+  await page.getByRole("menuitem", { name: "Gateway" }).click();
+  await page.getByRole("button", { name: "New Consumer" }).click();
+
+  await expect(page.getByRole("button", { name: "Create" })).toBeDisabled();
+  await page.getByPlaceholder("consumer-name.yaml").fill(filename);
+  await page.getByRole("button", { name: "Create" }).click();
+
+  await expect(
+    page,
+    "it creates the service and opens the file in the explorer"
+  ).toHaveURL(`/${namespace}/explorer/consumer/${filename}`);
+
+  /* fill in form */
+  await page.getByLabel("Username").fill("my-username");
+  await page.getByLabel("Password").fill("a-v3ry-g00d-pASSw0rd");
+  await page.getByLabel("Api key").fill("some-api-key");
+
+  const groupsElement = page.getByPlaceholder("Enter a group");
+
+  await Promise.all(
+    groups.map(async (group, index) => {
+      await expect(
+        groupsElement,
+        "it renders one input for every existing group +1 empty one"
+      ).toHaveCount(index + 1);
+
+      await groupsElement.last().fill(group);
+      await page
+        .locator("fieldset")
+        .filter({ hasText: "Groups (optional)" })
+        .getByRole("button")
+        .last()
+        .click();
+    })
+  );
```

Comment:
Very good catch. Did not think about, and now I'm surprised that it ever worked locally :D

Reply:
I think it is "safe" due to the way the test is structured. You first assert that there is the right amount of elements and then you fill in the last one. Thanks to this it will wait until the correct field is rendered.